common/flights/src/impls/kv_api_impl.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020-2021 The Datafuse Authors.
+//
+// SPDX-License-Identifier: Apache-2.0.
+//
+
+use common_exception::Result;
+use common_store_api::GetKVActionResult;
+use common_store_api::KVApi;
+use common_store_api::UpsertKVActionResult;
+
+use crate::GetKVAction;
+use crate::StoreClient;
+use crate::UpsertKVAction;
+
+#[async_trait::async_trait]
+impl KVApi for StoreClient {
+    async fn upsert_kv(
+        &mut self,
+        key: &str,
+        seq: Option<u64>,
```

Comment:
What does `seq` mean for an upsert operation?

What is the semantic of upsert, does it update the old value entirely if it exists?
My understanding of upsert is from this paper: http://supertech.csail.mit.edu/papers/BenderFaJa15.pdf

Why we are missing deletion?
If we want to use RocksDB as our backend, it has some advanced operations, e.g., multi-get, writebatch, range deletion. Would we consider them as well?

Reply:
- the `seq` is for compare-and-swap operations (a weak form of that ;)
  the following code snippets may help:
   https://github.com/datafuselabs/datafuse/blob/962232dc5be16566effc2beea517c82cbd6463ac/fusestore/store/src/meta_service/raftmeta.rs#L82-L89
   
   https://github.com/datafuselabs/datafuse/blob/962232dc5be16566effc2beea517c82cbd6463ac/fusestore/store/src/meta_service/state_machine.rs#L236-L267
   
- the `upsert` here is strickly less primitive than what you refer.
   I think it more like the semantic of `upsert` SQL,  roughly, "insert if not exists, otherwise update", but taking the sequence number into account as well.
   
- yes, in this PR, lots of primitive operations are missing, they are in the PR #881
  but batch write / deletion are still missing,  we should have them

Reply:
> If we want to use RocksDB as our backend

Maybe we should have that,  but not in v0.5

Reply:
Oh, thanks for the suggestion of utilizing the advanced operations of RocksDB, it really helps.

We've also considered that (and alternatives), but lots of things to be done in v0.5, we have to postpone it.